### PR TITLE
Fix `getToken()` iOS function for messaging

### LINKED ISSUE #2657: 🔥 Cloud Messaging "[messaging/unknown] The operation couldn’t be completed".
## Issue

Cloud messaging gives an error when using `messaging().getToken` after upgrading to V6 and RN 0.61:
```
Error: [messaging/unknown] The operation couldn’t be completed. (com.firebase.iid error 1001.)
NativeFirebaseError: [messaging/unknown] The operation couldn’t be completed. (com.firebase.iid error 1001.)
    at FirebaseMessagingModule.getToken (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:122099:28)
    at _callee2$ (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:117856:74)
    at tryCatch (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29289:19)
    at Generator.invoke [as _invoke] (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29464:24)
    at Generator.prototype.(anonymous function) [as next] (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29332:23)
    at tryCatch (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29289:19)
    at invoke (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29365:22)
    at http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:29395:13
    at tryCallTwo (http://localhost:8081/index.bundle?platform=ios&dev=true&minify=false:3223:7
```

---

### iOS

<details><summary>Click To Expand</summary>
<p>

#### `ios/Podfile`:

- [ ] I'm not using Pods
- [x] I'm using Pods and my Podfile looks like:

```ruby
platform :ios, '9.0'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'
target 'Glitz' do
  use_native_modules!
# use_frameworks!

  # Pods for Glitz
  pod 'FBLazyVector', :path => "../node_modules/react-native/Libraries/FBLazyVector"
  pod 'FBReactNativeSpec', :path => "../node_modules/react-native/Libraries/FBReactNativeSpec"
  pod 'RCTRequired', :path => "../node_modules/react-native/Libraries/RCTRequired"
  pod 'RCTTypeSafety', :path => "../node_modules/react-native/Libraries/TypeSafety"
  pod 'React', :path => '../node_modules/react-native/'
  pod 'React-Core', :path => '../node_modules/react-native/'
  pod 'React-CoreModules', :path => '../node_modules/react-native/React/CoreModules'
  pod 'React-Core/DevSupport', :path => '../node_modules/react-native/'
  pod 'React-RCTActionSheet', :path => '../node_modules/react-native/Libraries/ActionSheetIOS'
  pod 'React-RCTAnimation', :path => '../node_modules/react-native/Libraries/NativeAnimation'
  pod 'React-RCTBlob', :path => '../node_modules/react-native/Libraries/Blob'
  pod 'React-RCTImage', :path => '../node_modules/react-native/Libraries/Image'
  pod 'React-RCTLinking', :path => '../node_modules/react-native/Libraries/LinkingIOS'
  pod 'React-RCTNetwork', :path => '../node_modules/react-native/Libraries/Network'
  pod 'React-RCTSettings', :path => '../node_modules/react-native/Libraries/Settings'
  pod 'React-RCTText', :path => '../node_modules/react-native/Libraries/Text'
  pod 'React-RCTVibration', :path => '../node_modules/react-native/Libraries/Vibration'
  pod 'React-Core/RCTWebSocket', :path => '../node_modules/react-native/'
  # Uncomment the next line if you're using Swift or would like to use dynamic frameworks
  # use_frameworks!

  # Pods for Glitz
  pod 'React-cxxreact', :path => '../node_modules/react-native/ReactCommon/cxxreact'
  pod 'React-jsi', :path => '../node_modules/react-native/ReactCommon/jsi'
  pod 'React-jsiexecutor', :path => '../node_modules/react-native/ReactCommon/jsiexecutor'
  pod 'React-jsinspector', :path => '../node_modules/react-native/ReactCommon/jsinspector'
  pod 'ReactCommon/jscallinvoker', :path => "../node_modules/react-native/ReactCommon"
  pod 'ReactCommon/turbomodule/core', :path => "../node_modules/react-native/ReactCommon"
  pod 'Yoga', :path => '../node_modules/react-native/ReactCommon/yoga'

  pod 'DoubleConversion', :podspec => '../node_modules/react-native/third-party-podspecs/DoubleConversion.podspec'
  pod 'glog', :podspec => '../node_modules/react-native/third-party-podspecs/glog.podspec'
  pod 'Folly', :podspec => '../node_modules/react-native/third-party-podspecs/Folly.podspec'
  pod 'react-native-ble-plx', :path => '../node_modules/react-native-ble-plx'
  pod 'react-native-ble-plx-swift', :path => '../node_modules/react-native-ble-plx'
  pod 'react-native-radar', :path => '../node_modules/react-native-radar'

  # https://stackoverflow.com/questions/42021796/react-native-xcode-project-product-archive-fails-with-duplicate-symbols-for-arch
  # https://github.com/facebook/react-native/issues/12814


  pod 'react-native-ble-plx', :path => '../node_modules/react-native-ble-plx'

  pod 'RNCPushNotificationIOS', :path => '../node_modules/@react-native-community/push-notification-ios'

  post_install do |installer|
    installer.pods_project.targets.each do |target|
      targets_to_ignore = %w(React yoga)
      
      if targets_to_ignore.include? target.name
        target.remove_from_project
      end

      if target.name == 'react-native-ble-plx-swift'
        target.build_configurations.each do |config|
          config.build_settings['SWIFT_VERSION'] = '4.2'
        end
      end
      if target.name == 'react-native-config'
        phase = target.project.new(Xcodeproj::Project::Object::PBXShellScriptBuildPhase)
        phase.shell_script = "cd ../../"\
                            " && RNC_ROOT=./node_modules/react-native-config/"\
                            " && export SYMROOT=$RNC_ROOT/ios/ReactNativeConfig"\
                            " && export BUILD_DIR=$RNC_ROOT/ios/ReactNativeConfig"\
                            " && ruby $RNC_ROOT/ios/ReactNativeConfig/BuildDotenvConfig.ruby"

        target.build_phases << phase
        target.build_phases.move(phase,0)
      end
    end
  end

end
```

</p>
</details>

---

### Android

<details><summary>Click To Expand</summary>
<p>

#### Have you converted to AndroidX?

- [x] my application is an AndroidX application?
- [x] I am using `android/gradle.settings` `jetifier=true` for Android compatibility?
- [ ] I am using the NPM package `jetifier` for react-native compatibility?
</p></details>
---

## Environment

<details><summary>Click To Expand</summary>
<p>

**`react-native info` output:**

```
  SDKs:
    iOS SDK:
      Platforms: iOS 13.0, DriverKit 19.0, macOS 10.15, tvOS 13.0, watchOS 6.0
    Android SDK:
      API Levels: 23, 25, 26, 27, 28
      Build Tools: 27.0.3, 28.0.2, 28.0.3
      System Images: android-26 | Intel x86 Atom_64, android-27 | Intel x86 Atom_64, android-27 | Google APIs Intel x86 Atom, android-28 | Google APIs Intel x86 Atom
  IDEs:
    Android Studio: 3.2 AI-181.5540.7.32.5056338
    Xcode: 11.0/11A420a - /usr/bin/xcodebuild
  npmPackages:
    react: 16.9.0 => 16.9.0 
    react-native: 0.61.1 => 0.61.1 
  npmGlobalPackages:
    create-react-native-app: 2.0.2
    react-native-cli: 2.0.1
    react-native-git-upgrade: 0.2.7
    react-native-version: 3.1.0
```

- **Platform that you're experiencing the issue on**:
  - [x] iOS
  - [ ] Android
  - [ ] **iOS** but have not tested behavior on Android
  - [ ] **Android** but have not tested behavior on iOS
  - [ ] Both
- **`react-native-firebase` version you're using that has this issue:**
  - `6.0.0`
- **`Firebase` module(s) you're using that has the issue:**
  - `Cloud messaging`
- **Are you using `TypeScript`?**
  - `N`
  
</p>
</details>

### PR DESCRIPTION
**Pull request by anfriis**
This function often throws an exception:
`Cloud Messaging "[messaging/unknown] The operation couldn’t be completed"`
This fixes it and returns the FCM token successfully.

Fixes #2657 

### Checklist

- [x] Supports `Android`
- [x] Supports `iOS`
- [x] `e2e` tests added or updated in packages/**/e2e
- [x] Flow types updated
- [x] Typescript types updated

### Test Plan

I have tested on two different iPhone's, trying to get FCM token several times using this function, and it works now whereas it didn't before because it threw the exception above.

### Release Plan
[IOS][BUGFIX] [messaging] - Fix `getToken()` iOS function for messaging
 

[CATEGORY][type] [LOCATION] - Message

---

Think `react-native-firebase` is great? Please consider supporting the project with any of the below:

- 👉 Donate via [Open Collective](https://opencollective.com/react-native-firebase/donate)
- 👉 Follow [`React Native Firebase`](https://twitter.com/rnfirebase) and [`Invertase`](https://twitter.com/invertaseio) on Twitter
- 👉 Star this repo on GitHub ⭐️
- 👉 Contribute; see our [contributing guide](/CONTRIBUTING.md)

### CODE REVIEW

**Comment by mikehardy** [comment]
Tagging in @Salakar and @Ehesp - this could be a fix for what appears to me to be the hottest issue in the repo right now

**Comment by Salakar** [comment]
Thanks for the PR, but this looks incomplete to me, `authorisedEntity`, `scope` & `options` are no longer handled in the method (breaking change).

image

**Comment by anfriis** [comment]
Thanks for reviewing @Salakar. 
It may be a breaking change for someone whom it is currently working for. Though I assume that very few people (perhaps nobody) are using the advanced usage params `authorizedEntity` and `scope`, which defaults to `'FCM'`. and `options`. I am unfortunately not that deep into the internals of the framework to be able to change this.
By looking at the feedback on the issue #2657, the `getToken()` function is currently broken most of the time even for basic usage, so we should consider supporting basic usage first, and then support the more advanced usage later IMO.

**Comment by Ehesp** [comment]
@anfriis not sure that's a wise call on our end, we need it to be working with no breaking changes - who knows what people have done in their projects (patch packages, local fixes etc). We'll get it looked at asap.

**Comment by jacobp100** [comment]
What about if we call the function in this PR if the scope is undefined (defaulted to `*`) - and leave the current behaviour on other cases?

**Comment by MamyChow** [comment]
Any update ?
I don't understand why we don't use instanceIDWithHandler because firebase's docs say that's better to use it unless there is a need to enable multiple senders.

**Comment by ottob** [comment]
This fixed getToken for me too. And then with #3075 I could finally get my notifications.

Thanks @anfriis!

File: packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m [hunk 1]
```diff
@@ -96,11 +96,11 @@ - (NSDictionary *)constantsToExport {
     options = @{@"apns_token": [FIRMessaging messaging].APNSToken};
```
Comment (Gray-Wind) [inline packages/messaging/ios/RNFBMessaging/RNFBMessagingModule.m]: You should also drop options, as they are not used anymore.

**Comment by Ehesp** [comment]
Thanks for sending this PR up. I'm going to close it as we ended up re-writing a lot of the iOS internals as part of wider fixes on https://github.com/invertase/react-native-firebase/pull/3339. We implemented this change along with supporting scopes/auth-entities too using the previous API.

**Pull request closed without merging**